packages/venia-concept/src/components/CreateAccount/createAccount.js

```diff
@@ -0,0 +1,220 @@
+                    />
+
+                    <Input
+                        onChange={this.updateLastName}
+                        label={'Last Name'}
+                        required={true}
+                    />
+
+                    <Input
+                        onChange={this.updatePassword}
+                        label={'Password'}
+                        type={'password'}
+                        required={true}
+                        placeholder={'Enter a password'}
+                        helpText={
+                            'Password must be at least 8 characters long and contain 3 or more of the following: Lowercase, Uppercase, Digits, or Special Characters. (ex. Password1)'
+                        }
+                    />
+
+                    <Input
+                        onChange={this.updatePasswordConfirm}
+                        label={'Confirm Password'}
+                        type={'password'}
+                        required={true}
+                        placeholder={'Enter the password again'}
+                        helpText={passwordHelpText}
+                        helpType={passwordHelpType}
+                    />
+
+                    <Checkbox
+                        label={'Subscribe to news and updates'}
+                        select={this.handleCheckboxChange}
+                        initialState={this.state.subscribe}
+                    />
+                    <div className={classes.createAccountButton}>
+                        <Button type="submit" disabled={disableAccountCreation}>
+                            Create Account
+                        </Button>
+                    </div>
+                    {errorMessage}
+                </Form>
+            </div>
+        );
+    }
+
+    onCreateAccount = () => {
+        if (!this.disableAccountCreation) {
+            const newCustomer = {
+                customer: {
+                    firstname: this.state.firstName,
+                    lastname: this.state.lastName,
+                    email: this.state.email
+                },
+                password: this.state.password
+            };
+            this.props.createAccount(newCustomer);
+        }
+    };
+
+    handleCheckboxChange = value => {
+        this.setState({ subscribe: value });
+    };
+
+    checkEmail = debounce(async email => {
+        try {
+            const body = {
+                customerEmail: email,
+                website_id: null
+            };
+            const response = await request(
+                '/rest/V1/customers/isEmailAvailable',
+                {
+                    method: 'POST',
+                    body: JSON.stringify(body)
+                }
+            );
+            this.setState({ emailAvailable: response, checkingEmail: false });
+        } catch (error) {
+            console.warn('err: ', error);
+        }
+    }, 300);
+
+    updateLastName = newLastName => {
```

Comment:
Nice job on these methods. It's a really good idea to keep `setState` itself behind some business logic.